\documentclass[12pt]{article}
\usepackage{graphicx}
\usepackage{float}
\usepackage{subcaption}
\usepackage{hyperref}
\usepackage{mathtools}
\usepackage[usenames,dvipsnames]{xcolor}
\usepackage[authoryear]{natbib}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{bigints}
\usepackage{array}
\usepackage{tikz}
\usepackage{longtable}
\usepackage{geometry}

\bibliographystyle{plainnat}

\setlength{\parskip}{\baselineskip}
\setlength{\parindent}{0pt}

\newcommand{\e}[1]{{\mathbb E}\left[ #1 \right]}
\newcommand{\given}{\mid}
\newcommand{\secref}[1]{``\nameref{#1}''}
\newcommand{\tri}{\textit{trichocarpa}} 
\newcommand{\bals}{\textit{balsamifera}}

\include{macros}

\newcommand{\gb}[1]{{\it\color{magenta}{(#1)}}}
\newcommand{\plr}[1]{{\it\color{purple}{(#1)}}}
\newcommand{\gc}[1]{{\it\color{blue}{(#1)}}}

\geometry{a4paper}

\title{Inferring Continuous and Discrete Population Genetic Structure Across Space}
\date{\vspace{-5ex}}
\author{
Gideon S. Bradburd$^{1,a}$, 
Graham M. Coop$^{2,b}$,
Peter L. Ralph$^{3,c}$}

\begin{document}

\maketitle

\textsuperscript{1}
Department of Integrative Biology, 
Ecology, Evolutionary Biology, and Behavior Graduate Group,
Michigan State University, MI 48824

\textsuperscript{2}
Center for Population Biology,
Department of Evolution and Ecology, 
University of California, Davis, CA 95616

\textsuperscript{3}
Institute of Ecology and Evolution,
Department of Mathematics,
University of Oregon, Eugene, OR 97403

\textsuperscript{a}bradburd@msu.edu; 
\textsuperscript{b}gmcoop@ucdavis.edu;
\textsuperscript{c}plr@uoregon.edu\\\\\

\newpage
 

\begin{abstract}
One of the classic problems in population genetics is the characterization 
of discrete population structure when the genotyped samples also show 
continuous patterns of genetic differentiation.
Especially when sampling is discontinuous, 
clustering or assignment methods may incorrectly ascribe differentiation 
due to continuous processes (e.g., isolation by distance) 
to discrete processes, such as geographic, ecological, or reproductive barriers 
between populations.
This is partly a result of the difficulty of sampling uniformly and continuously 
across the range of a population or species, 
but more, it reflects a shortcoming of current methods for inferring and 
visualizing population structure from genetic data in the face of data 
that are characterized by both continuous and discrete population structure.
Here, we present a novel statistical framework for the simultaneous inference 
of continuous and discrete patterns of population structure.
The method estimates ancestry proportions for each 
sample from a set of discrete population layers, 
and, within each layer, estimates a rate at which relatedness decays with distance.
This model explicitly addresses the ``clines vs. clusters" problem in 
modeling population genetic variation by jointly accommodating both 
continuous and discrete patterns of differentiation. 
We demonstrate the utility of this approach using both simulations and an empirical application.
\end{abstract}


\newpage
\section*{Introduction}
A fundemental quandary in the description of biological diversity, is
the fact that diversity shows both discrete and continuous patterns. 
For example, there is disagreement on species concepts because the
process of speciation is usually gradual, and so there is no set point
in the continuous divergence of populations when they become discrete species.
This issue extends below the species level as patterns of phenotypic
and genetic diversity within and among populations are shaped
by continuous migration and drift, as well as more discrete events and
processes such as geographic barriers, rapid expansions, bottlenecks, and rare
long distance migration. 

From a practical standpoint, even while acknowledging continuous
processes, we often need to identify somewhat separable populations 
from which individuals are sampled \citep{wright1949genetical}.
Delineating populations is useful for systematics and for
informing conservation priorities \citep{Moritz1994,Waples_1998,Moritz_etal_2002}.
Furthermore, we often need to identify sets of indivduals resulting from reasonably coherent
evolutionary histories for downstream analyses to learn about population history and adaptation. 

%In studying populations we need group individuals 
%Many downstream 
 
%Biology diversity 


%The fundamental goal of evolutionary genetics is to link patterns of genetic variation 
%to the processes that have generated and maintained them.
%Often, the first step in the analysis of genetic data is to define evolutionary units of analysis
%by delineating discrete populations from which individuals or alleles are sampled \citep{wright1949genetical}.
%These units are useful for defining a scope of study in research on selection or local adaptation (ref),
%or for identifying either locally adapted alleles (human skin color, ralph \& coop, gartersnakes) or alleles involved in disease (cystic fibrosis).
%Delineating populations can also be useful for informing conservation priorities \citep{Moritz1994,Waples_1998,Moritz_etal_2002}.
%However, if underlying population structure is continuous, this may entail a loss of power.

There have been many methods proposed to characterize population
genetic structure,
including generating population phylogenies \citep{CavalliSforza1975, treemix},
$k$-means clustering analyses performed on dimensionality-reduction approaches (ref), 
such as principal components analysis \citep{menozzi1978synthetic,novembre_interpreting_2008, price2006eigenstrat},
and model-based clustering approaches \citep[e.g.][]{STRUCTURE, falush2003, hubisz2009,ADMIXTURE, FINESTRUCTURE, fastStructure, huelsenbeck2007inference, Corander2003,TESS,geneland}, among others.
Here, we will focus on model-based clustering, the most widely used class of approaches for population delineation.
Like the method we present here, these are based on a statistical model,
but do not perform inference directly with a demographic model.
Each of these methods perform best in particular situations,
but all can give misleading results when applied to data that show a continuous pattern of isolation by distance 
\citep{Wright1943, novembre_interpreting_2008, Frantz2009}.

A pattern of isolation by distance (IBD) arises when migration in a population is spatially limited \citep{Wright1943},
and has been modeled in both spatially continuous frameworks \citep{malecot1969mathematics,Slatkin1985}, 
in which an individual's parents are found within some radius of that individual's location,
and discrete, ``stepping-stone'' frameworks \citep{kimura1964stepping},
in which migration occurs only between neighboring demes.
Given the generality of the circumstances that generate a pattern of isolation by distance, 
it is unsurprising that IBD is very widespread in nature \citep{meirmans2012,Sexton_etal_2014}.

The ubiquity of isolation by distance presents a challenge for models of discrete population structure,
as it is frequently difficult to determine whether observed patterns of genetic variation are 
continuously distributed across a landscape, or instead are partitioned in discrete clusters.
This problem can be compounded if sampling is done unevenly or discretely across a population or species' range,
and has given rise to a debate in the population genetic literature about whether 
individuals are best described as being drawn from continuous clines or discrete clusters 
\citep[e.g.][]{SerrePaabo2004,rosenberg2005clines}.
%As a further complication, if local genetic drift is strong,
%continous models are expected to give rise to spatially patchy patterns
%\citep{SLFV,bramson1989crabgrass}.

Existing model-based clustering approaches can only describe continuous patterns of variation using
discrete clusters, and so tend to erroneously describe continuously distributed variation with multiple clusters that 
show spatially autocorrelated cluster membership \citep{Frantz2009,meirmans2012}.
In analyses of empirical datasets, which often are characterized by IBD,
model-based clustering approaches will therefore tend to overestimate
the number of discrete clusters present. 

%with downstream implications for interpretation of results.
%For example, conservation resources might be spread too thinly over areas 
%that are spuriously assumed to be discrete management units.
%Or, in studies of locally adaptive loci, or of the genetics of disease,
%discrete populations may be modeled in separately,
%reducing the global power to detect the genomic basis of complex phenotypes.

To address this, we set out to develop
a model-based clustering method that when possible uses isolation by distance 
to explain observed genetic variation.
With an explicit spatial component, discrete population structure need only be invoked when genetic differentiation 
in the data is deviates significantly from that expected given geographic separation.

In this paper, we present a statistical method for simultaneously 
modeling continuous and discrete patterns of population structure.
We model genetic variation in genotyped individuals as 
partitioned within or admixed across a specified number of discrete layers,
within each of which relatedness decays as a parametric function of the distance between samples.
We also implement a cross-validation approach for comparing and selecting models across different numbers of layers,
and we demonstrate the utility of our approach using both simulated and empirical data.
The implementation of this method, \texttt{conStruct}, is available as an R package at 
\href{https://github.com/gbradburd/conStruct}{https://github.com/gbradburd/conStruct}.

\section*{Methods}

\paragraph{Data}
The statistical framework of our approach is conceptually similar to that in \cite{Wasser2004}, \cite{BEDASSLE}, and \cite{spacemix},
although we use a somewhat different summary statistic than in this previous work.
The model works with allele frequencies at $L$ unlinked, bi-allelic single nucleotide polymorphisms (SNPs) genotyped across $N$ samples.
Each ``sample'' may be a collection of diploids, or frequencies estimated from pooled sequencing.
We write $N_i$ for the number of chromosomes sequenced in the $i^\text{th}$ sample.
The \emph{sample frequency} at locus $\ell$ in sample $n$, denoted $f_{n,\ell}$, 
is calculated by first arbitrarily choosing one of the observed alleles at locus $\ell$ to count, 
then dividing the number of observations of that counted allele by the total number of chromosomes genotyped at that locus
in sample $n$.
The choice of allele does not affect subsequent calculations, and so may be arbitrary.
We then calculate the \emph{allelic covariance} between samples $i$ and $j$, denoted $\widehat{\Omega}_{i,j}$,
as the expected covariance of distinct individual alleles chosen from each of the two samples at a random locus,
choosing a random allele to code as `1' (and the other as `0').
Concretely, this is
\begin{equation}
\widehat{\Omega}_{i,j} = 
    \frac{1}{L} \sum_{\ell=1}^L \left( (f_{i,\ell}-1/2) (f_{j,\ell}-1/2) - \delta_{i,j} f_{i,\ell}^2 / (N_i-1) \right),
\label{allelic_covariance}
\end{equation}
where $\delta_{i,j}=1$ if $i=j$ and is 0 otherwise.
Although we describe this as a covariance between individually drawn alleles,
$\widehat{\Omega}_{i,j}$ is in fact also the covariance between the allele frequencies
of a randomly chosen allele in samples $i$ and $j$, as long as $i \neq j$.
The diagonal (where $i=j$) does not have this interpretation.

In addition to these differences on the diagonal,
this differs from the usual ``genetic covariance'' \citep{mcvean_genealogical_2009}
in that (a) we do not subtract locus means,
and so to make the statistic invariant under choices of reference allele, 
(b) we randomly choose an allele to count at each locus.
We discuss the derivation of Eqn. \eqref{allelic_covariance} further in the Appendix (\secref{allelic_cov}).
As noted in \cite{EEMS},
this covariance has a close relationship to the pairwise genetic distance:
if $D_{i,j}$ is the mean density of sites at which random samples from $i$ and $j$ differ at a randomly chosen locus,
then $\Sigma_{i,j} = (1 - 2 D_{i,j})/4$.
Therefore, unlike genetic covariance,
this allelic covariance is affected in shape by singleton sites,
so it may be advisable to filter these prior to analysis
if these are likely to contain a large percentage of errors.

\paragraph{Continuous and discrete differentiation}
Clustering approaches to describing genetic variation
are useful because population history can often be meaningfully described on a coarse scale
by interactions between discrete ``populations''
whose relationships are delimited by patterns of glaciation,
large-scale migration,
mountain ranges, and the like.
Here we add a spatial component within each such discrete historical component,
which we refer to as a set of ``layers'' which overlay the modern map.
We imagine each layer as a geographically distributed population,
and that each sample is composed of a mixture of contributions from each of these layers.
These layers thus take the place of ``clusters'' or ``populations'' in clustering methods,
but we do not adopt these terms as they connote a lack of geographical separation within each.

Within each of these layers,
allele frequencies have positive covariance at geographically close locations,
but this covariance is allowed to decay towards zero as geographic distance increases.
This pattern of spatial decay reflects how migration between neighboring demes  
homogenizes allele frequency changes that arise locally due to drift, 
but this mixing less effectively homogenizes geographic distant demes,
resulting in a continuous pattern of isolation by distance within each cluster.
There is a fixed amount of covariance between layers, irrespective of spatial location.
Within each layer allele frequencies are expected to change gradually with distance,
but observed frequencies can change abruptly at many loci 
if the proportions by which individuals' ancestry derive from each layer 
(the ``admixture'' coefficients) 
do so as well.

To allow flexibility in the form of the decay of allelic covariance with geographic distance within each cluster, 
we define the covariance within cluster $k$ between samples $i$ and $j$ to be:
\begin{equation}
G^{(k)}_{i,j} 
    = 
    \alpha^{(k)}_0 \left( \text{exp} \left( -(\alpha^{(k)}_D D_{i,j}) ^ {\alpha^{(k)}_2}	\right) \right) + \phi^{(k)}
\label{within_cluster_covariance}
\end{equation}
where the superscript $(k)$ denotes parameters specific to the $k$th cluster.
The quantity $D_{i,j}$ is the observed geographic distance between samples $i$ and $j$,
the $\alpha^{(k)}$ parameters control the shape of the decay of covariance with distance in the cluster,
and $\phi^{(k)}$ is a parameter that describes the background covariance within the cluster. 
If two samples draw 100\% of their ancestry from cluster $k$, then their covariance under the model is $G^{(k)}_{i,j}$;
if they are furthermore geographically very close ($D_{i,j}=0$)
they will have covariance $\alpha^{(k)}_0 +  \phi^{(k)}$.
If the geographic distance between them is very large, 
their covariance will be equal to to the background level $\phi^{(k)}$ within the cluster.
The ``shared drift'' parameter $\phi^{(k)}$ can be interpreted as the branch length 
connecting the $k$th population to the population ancestral to all modeled
layers \citep{patterson_ancient_2012, peter_fstats}.

We then allow samples to draw their ancestry from more than one cluster.
The ``admixture'' proportion of the $i$th sample in the $k$th cluster, denoted $w^{(k)}_i$,
gives the proportion of alleles from sample $i$ that derive from
cluster $k$ (and so $\sum_K w^{(k)}_i =1$).
A visual representation of the method mechanics is shown in Fig \ref{schematic}.

\begin{figure}
	\centering
		{\includegraphics[width=4in,height=2.5in]{figs/schematic/method_schematic.png}}
		\caption{Schematic of our method, using $K=3$ as an example.
			     The pattern of isolation by distance within each layer is denoted by the color gradient in each of the layer ellipses, 
			     and $\phi_{(k)}$ denotes the covariance shared by samples with ancestry entirely in the $k$th cluster.
			     Sampled populations on the landscape are inferred to be admixed between these layers; 
			     the $i$th sample draws proportion $w^{(k)}_i$ of its ancestry from cluster $k$.
			    }\label{schematic}
\end{figure}

We can then describe the covariance between samples $i$ and $j$ across all layers, $\Omega_{i,j}$,
by summing their within-cluster spatial covariances ($G_{i,j}^{(k)}$ in cluster $k$) across all $K$ layers,
weighted by the relevant admixture proportions.
\begin{equation}
\Omega_{i,j} = \gamma + \sum\limits_K w^{(k)}_i w^{(k)}_j
G^{(k)}_{i,j} + \delta_{i,j}\eta_i .
\label{cross_cluster_covariance}
\end{equation}
In this equation, $w^{(k)}_i w^{(k)}_j$ is the proportion of alleles that \emph{both}
sample $i$ and sample $j$ have inherited from cluster $k$.

In addition to the admixture-weighted sum of the within-cluster spatial covariances,
this function contains two terms, $\gamma$ and $\delta_{i,j}\eta_i$.
The first, $\gamma$, describes the global allelic covariance between all samples, 
and arises because all samples share an ancestral mean allele frequency at each locus,
which generates a base-line covariance.
In the final term, $\delta_{i,j}$ is an indicator variable that takes a value of 1 when $i$ equals $j$ and 0 otherwise,
and $\eta_i$ adds variance specific to sample $i$.
This term on the diagonal of the parametric covariance matrix captures processes shaping variance within the sampled deme, 
such as inbreeding and the sampling process.

\paragraph{Likelihood and inference}
If allele frequencies at each locus
were independent between loci and multivariate normally distributed across populations, 
then their allelic covariance $\widehat{\Omega}$ 
would be Wishart distributed with degrees of freedom equal to $L$, 
the number of loci genotyped.
We use this as a convenient approximation to the true distribution described above,
and so define the likelihood of the allelic covariance to be
\begin{equation}
P(\widehat{\Omega} \; | \; \Omega) = \mathcal{W} \left( L\widehat{\Omega} \; | \; \Omega,L\right) .
\end{equation}
Linkage disequilibrium (LD) between loci will decrease the effective number of degrees of freedom, 
a complication we discuss further below.

We estimate the values of the parameters of the model using a Bayesian approach.
Acknowledging the dependence of the parametric covariance matrix $\Omega$ on its constituent parameters
$w,\alpha,\phi,\eta,\gamma$ and on the (observed) geographic distances $D$ 
with the notation $\Omega(w,\alpha,\phi,\eta,\gamma,D)$,
we denote the posterior probability density of the parameters as:
\begin{equation}
P\left( w,\alpha,\phi,\eta,\gamma \;	| \; \widehat{\Omega}, L \right) \propto
P\left(\widehat{\Omega} \; | \; \Omega(w,\alpha,\phi,\eta,\gamma,D) \right)
P(w)P(\alpha)P(\phi)P(\eta)P(\gamma) ,
\end{equation}
where $P(w)$, $P(\alpha)$, $P(\phi)$, $P(\eta)$, and $P(\gamma)$, are prior distributions.
All parameters are given (half-)Gaussian priors except for $\alpha_2$, which is uniform on $(0,2)$ and $w$, which is Dirichlet
(see Table \ref{tab:param_prior_tab} for specifics).
Parameters are independent between layers.
We use Hamiltonian Monte Carlo as implemented in STAN \citep{stan, NUTS, stan_lib, rstan} 
to estimate the posterior distribution on the parameters.
We also provide an R package \citep{R} called \texttt{conStruct} that functions as a wrapper 
around this inference machinery.


\paragraph{Relationship of this model to nonspatial structure models}

A nice feature of our approach is that the model described in Eqn \eqref{cross_cluster_covariance} 
contains a nonspatial assignment model as a special case 
(see \secref{model_app} for a more in-depth discussion). 
By setting $\alpha_0^{(k)}$ to zero for all $k$, 
we obtain a nonspatial model in which each cluster has its own allele frequency at each SNP, 
and individuals draw a proportion of their ancestry from each cluster. 
This model is strongly conceptually similar to that of STRUCTURE \citep{STRUCTURE} 
and related models \citep[e.g.][]{ADMIXTURE}; 
the only difference is that our likelihood assumes that allele frequencies are normally distributed 
around their expectations, 
while the standard assignment methods assume that the error is binomial distributed (see XXXX).  
Therefore, we can compare the fit of the different models 
--- spatial vs. nonspatial, across different values of $K$ --- 
by comparing their performance in a common framework.

\paragraph{Choice of cluster number and cross-validation}
There are a number of reasons, both statistical and biological, 
why there is no true (or right) number of layers for real datasets,
discussed further in the Discussion.
However, it is still important to assess whether additional spatial layers (larger $K$)
meaningfully model pattern in the data
or merely explain spurious variation introduced by noise
-- in other words, whether additional model complexity
provides significant explanatory power.
Toward that end, we have implemented a method for 
statistically comparing \texttt{conStruct} results across different values of $K$ 
and between the spatial and nonspatial models.

Several approaches have been used as model choice criteria 
for the number of clusters in population genetic data, including: 
the use of information theoretic criteria \citep[e.g.][]{ADMIXTURE}, such as AIC or BIC;
comparisons of the likelihood of the data across different values of $K$, 
with various criteria on how to choose a single value (e.g., \citep{Evanno2005});
or comparisons of the marginal likelihood, 
generated either via various approximations \citep[e.g.][]{STRUCTURE}
or via a more rigorous and computationally intensive approach such as thermodynamic integration \citep{verity_nichols2016}
or inference using a Dirichlet process prior \citep{huelsenbeck2007inference}
See \cite{verity_nichols2016} for a discussion of these approaches and comparison
between several methods.

We choose to take a cross-validation approach \citep[similar in spirit to][]{ADMIXTURE_xval}.
We estimate the posterior on the parameters of the model using a ``training" partition of the data 
(in practice the training set is a random subset of $90\%$ of the loci). 
We then calculate the likelihood of the ``testing" subset (the remaining $10\%$ of the loci), 
averaged over the posterior of the parameters. 
We repeat this procedure a number of times, 
take an average of the relative log-likelihoods over the runs at each $K$, 
and select the model with the significantly best predictive accuracy
(see \secref{Xvalidation} for more on our cross-validation procedure).
This approach rewards models with larger values of $K$, 
which increase the likelihood of the training partition, 
as they fit the data better,
but the improvement in the likelihood will plateau (or peak), 
as above a certain $K$ the model is fitting noise specific to the training data 
and therefore not fitting the testing set any better.
\plr{I wonder if we can just cite \cite{verity_nichols2016} rather than this laundry list here?}

We use cross-validation to attack this problem.
To do this,
we use a ``training" partition of the data (in practice, a random 90\% subset of the loci)
to estimate the posterior distribution of the parameters,
and then calculate the log-likelihood of the remaining ``testing'' loci,
averaged over the posterior.
Prediction accuracy of a particular model (e.g., value of $K$)
is then measured using this log-likelihood,
averaged over a number of independent data partitions.
The best model is judged to be the simplest one with significantly better predictive accuracy
than others (see \secref{Xvalidation} for more on our cross-validation procedure).
In general, larger values of $K$ allows the model more flexibility,
and thus increases the likelihood of the training partition, 
but this improvement in the likelihood will plateau (or even peak), 
since above a certain $K$ the model only fits noise specific to the training data 
rather than generalizable patterns.
At any value of $K$, support for the spatial model over the nonspatial model 
means that isolation by distance is likely a feature of the data.

\gb{something about how we choose the best model?}
\plr{how's this?}
Cross-validation provides a valuable summary of how much explanatory power
is added by spatial structure within each layer, and each additional layer.
A simple $t$-test can be used to assess whether additional layers
show statistically significantly greater predictive accuracy.
However, we remind users that ``statistical significance does not imply real-world significance'',
and so small but statistically significant differences between models
should probably not be relied on too strongly.


\section*{Results}

\subsection*{Simulations}
We first test the method with scenarios generated using the coalescent simulator \texttt{ms} \citep{Hudson2002}.
To generate discrete population structure, we simulate $K$ distinct populations,
each of which split from a common ancestor $\tau_{\text{s}}$ units of coalescent time in the past,
without subsequent migration between them.
To generate continuous differentiation within each cluster,
at time $\tau_{\text{e}}$ in the past,
each of these discrete clusters instantaneously colonizes a lattice of demes,
for which we use a stepping stone model with symmetric migration 
to nearest neighbors (eight neighbors, including diagonals).
Finally, at time $\tau_{\text{a}}$ in the past 
we generate a single dataset 
by collapsing those $K$ discrete lattices into a single grid of demes that are admixed to various degrees from these different layers.
(see Fig \ref{sim_setup}).
We simulated datasets using $K=1$, 2, and 3 layers,
in each sampling 10,000 unlinked loci from each of 20 haploid individuals from every deme.

We ran \texttt{conStruct} analyses on each of these simulated datasets, 
specifying both a spatial and nonspatial model for $K = $ 1 through 7, 
then used cross-validation to compare the predictive performance of the models.
In Fig \ref{simK1_pies}, we show the results of the nonspatial model 
run with between 1 and 7 clusters, applied to the dataset simulated under $K=1$,
and in Fig \ref{sim_xvals} we show the results of the cross-validation analysis for each of the three simulated datasets.
The admixture pie plots for the spatial and nonspatial models run on the datasets simulated under $K=2$ and $K=3$
are shown in Figs \ref{simK1_nsp_pies}-\ref{simK3_sp_pies}.
To confirm that results from our nonspatial model were qualitatively similar to those of 
other nonspatial model-based clustering algorithms, 
we also analyzed each simulated dataset using fastSTRUCTURE \citep{fastStructure}, 
inferring admixture under models with $K=2$ through 4.

A number of results are immediately clear from these tests with simulated data.
First, \texttt{conStruct} performs well on all simulated scenarios.
Across all values of K and for all simulations, 
the spatial model is correctly preferred over the nonspatial model, 
and it infers the true admixture proportions with high accuracy, tight precision, and good coverage 
(Fig \ref{sim_xvals}).
Second, the predictive accuracy of the nonspatial model increases 
with the number of clusters in the model, 
as it describes continuous, spatial patterns of relatedness between demes within the same cluster 
using gradients of admixture across more and more discrete clusters.
This behavior of the nonspatial model is highlighted in 
the analyses of data simulated under $K=1$ (Fig \ref{simK1_pies} and \ref{simK1_nsp_pies}):
as $K$ in the model increases, 
the sampled area is increasingly divvied up between clusters 
with ``centers of mass" in the $K$ geographical extremes of the simulated space.
At $K=2$, the clusters are centered in opposite corners on the map; 
at $K=3$, in the lower two corners and the top half; 
at $K=4$, all four corners, and so on.
These results are qualitatively similar to those generated using fastSTRUCTURE \citep{fastStructure}, 
which also describes continuous differentiation using admixture between clusters
 (Fig \ref{fastStr_simK1}-\ref{fastStr_simK3}).
While it is true that each such cluster is more similar within itself
than it is to other clusters,
we know that these boundaries are arbitrary,
since the data were simulated with no such boundaries.

In contrast, the predictive accuracy of the spatial \texttt{conStruct} model
either plateaus at or declines from the true value of $K$ used to simulate the data (Fig \ref{sim_xvals}, second column).
And, as $K$ in the spatial model increases, 
deme membership in layers beyond the number used to simulate the data does not appreciably increase 
(Fig \ref{simK1_sp_pies}, \ref{simK2_sp_pies}, \ref{simK3_sp_pies}).
That is, even when the spatial model is run with $K=7$, 
the inferred admixture proportions are nearly identical to 
those estimated under the true value of $K$ for each simulation.

\gb{could talk about proportion of relatedness explained by each layer here.
e.g., Another way to visualize the difference in the behavior of the spatial and nonspatial models 
is to summarize the proportion of relatedness between samples explained by each cluster...}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK1_nsp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK1_nsp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK1_nsp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K4.pdf}}
		\subcaptionbox{$K=2$\label{simK1_sp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K5.pdf}}
		\subcaptionbox{$K=3$\label{simK1_sp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K6.pdf}}
		\subcaptionbox{$K=4$\label{simK1_sp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using the nonspatial model for $K=2$ through 4 (top row)
	and also the spatial model for $K=2$ through 4 (bottom row).
	The data were simulated using a single cluster with nearest-neighbor symmetric migration between demes.
    }\label{simK1_pies}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=8.5in]{figs/sims/xvals.pdf}}
	\caption{
	Cross-validation results for data simulated under $K=1$, $K=2$, and $K=3$, 
	comparing the spatial and nonspatial \texttt{conStruct} models run with $K=1$ through 7.  
	The first panel in each row shows all results; 
	the second panel shows results for just the spatial model.
	The inset plots zoom in on cross-validation results outlined in the dotted boxes.
    }\label{sim_xvals}
\end{figure}

\subsection*{Empirical Applications}
To demonstrate the utility of this method, we analyzed empirical population genomic data from two systems:
a contact zone between two poplar species in northwestern North America,
and a large North American sample of black bears.

\subsubsection*{Poplars}

Trees in the genus \textit{Populus} (poplars, aspens, and cottonwoods) 
are distributed throughout the Northern Hemisphere;
species in the genus regularly co-occur and, 
where they do, they frequently hybridize \citep{eckenwalder1984, Cronk2005}.

Here, we focus on two \textit{Populus} species in the Pacific Northwest: 
\textit{Populus trichocarpa}, the black cottonwood,
and \textit{Populus balsamifera}, the balsam poplar.
These species have a broad zone of overlap and are hypothesized to hybridize \citep{geraldes_etal_2014, suarezgonzalez_etal_2016}.
Both species are sampled over a large geographic region, 
and show spatial patterns of genetic and phenotypic variation \citep{slavov_etal_2012, mckown_etal_2013},
making the system well-suited to test an analysis designed to describe
how genetic structure is partitioned both discretely across population (or species) clusters
and continuously with geographic distance.
We organize the results of our analyses around the following questions: 
\begin{enumerate}
\item Given the known history of hybridization between this pair of poplar species, 
is there support for two discrete clusters, 
rather than a single cline of ancestry across their overlap zone?
\item If there is evidence for discrete population structuring, 
does it fall solely along species boundaries, 
or is there sub-structuring between different populations of the same species?
\item Is there support for a pattern of isolation by distance within
the inferred clusters, and, if so, 
does it differ between different layers?
\end{enumerate}

We address these questions using a dataset generated and originally published by \cite{geraldes_etal_2014}.
The published data consist of 434 individuals sampled from 35 drainages 
and genotyped at just over 33,000 loci (map of the sampling shown in Fig \ref{populus_map}).
The data were generated using RAD-seq, 
and showed a strong pattern of bias in allelic dropout 
(the majority of missing data were from drainages with only \textit{Populus balsamifera} individuals).
To ameliorate some of the problems that arise when there is a strong bias in which data are missing, 
we dropped loci for which any data were missing, 
resulting in just over 20,200 loci retained for analysis.  
We then analyzed these data, grouped by drainage, using both the spatial and nonspatial \texttt{conStruct} models, 
using $K = 1$ through 7,
and compared these models using cross-validation.
The results of these analyses are shown in 
Figs \ref{populus_xvals}--\ref{populus_nsp_clst_covs}. %,populus_nsp_pies,populus_sp_clst_covs,populus_nsp_clst_covs


\begin{figure}
	\centering
		{\includegraphics[width=6in,height=5in]{figs/populus/populus_sampling_map.pdf}}
	\caption{
	Map of the sampled \textit{Populus} populations included in the analysis.
	The color of the sampling location denotes the putative species.
    }\label{populus_map}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=3in]{figs/populus/populus_std_xval.pdf}}
	\caption{
	Cross-validation results for \textit{Populus} dataset 
	comparing the spatial and nonspatial \texttt{conStruct} models run with $K=1$ through 7.  
	The first panel in each row shows all results; 
	the second panel zooms in on the results from analyses run with $K = 2$ through 7.
    }\label{populus_xvals}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{populus_sp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp2.pdf}}
		\subcaptionbox{$K=3$\label{populus_sp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp3.pdf}}
		\subcaptionbox{$K=4$\label{populus_sp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp4.pdf}}
		\subcaptionbox{$K=5$\label{populus_sp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp5.pdf}}
		\subcaptionbox{$K=6$\label{populus_sp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp6.pdf}}
		\subcaptionbox{$K=7$\label{populus_sp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_sp7.pdf}}
	\caption{
	Map of admixture proportions estimated for the \textit{Populus} dataset 
	using the spatial model for $K=2$ through 7.
    }\label{populus_sp_pies}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{populus_nsp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp2.pdf}}
		\subcaptionbox{$K=3$\label{populus_nsp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp3.pdf}}
		\subcaptionbox{$K=4$\label{populus_nsp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp4.pdf}}
		\subcaptionbox{$K=5$\label{populus_nsp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp5.pdf}}
		\subcaptionbox{$K=6$\label{populus_nsp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp6.pdf}}
		\subcaptionbox{$K=7$\label{populus_nsp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/populus/populus_nsp7.pdf}}
	\caption{
	Map of admixture proportions estimated for the \textit{Populus} dataset 
	using the nonspatial model for $K=2$ through 7.
    }\label{populus_nsp_pies}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=4in]{figs/populus/populus_sp_clst_covs.pdf}}
	\caption{
	Plots showing the cluster-specific parametric covariance IBD-curves 
	estimated for the \textit{Populus} data using 
	the spatial \texttt{conStruct} model run with $K=2$ through 7.
	Line colors are consistent with cluster colors in Fig \ref{populus_sp_pies}.
	Points are colored by the species they are a covariance between:
	black on black points are sample covariances between populations of \textit{Populus balsamifera};
	green on black points are sample covariances between \bals{} and \tri{};
	green on green points are sample covariances between \tri{} and \tri{}.
    }\label{populus_sp_clst_covs}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=4in]{figs/populus/populus_nsp_clst_covs.pdf}}
	\caption{
	Plots showing the cluster-specific parametric covariance IBD-curves 
	estimated for the \textit{Populus} data using 
	the nonspatial \texttt{conStruct} model run with $K=2$ through 7.
	Line colors are consistent with cluster colors in Fig \ref{populus_nsp_pies}.
	Points are colored by the species they are a covariance between:
	black on black points are sample covariances between populations of \textit{Populus balsamifera};
	green on black points are sample covariances between \bals{} and \tri{};
	green on green points are sample covariances between \tri{} and \tri{}.
    }\label{populus_nsp_clst_covs}
\end{figure}

All models with $K>1$ assigned the majority of each species
to distinct layers, with some populations drawing ancestry from both layers.
This provides strong support for discrete population structure between the two species,
with some admixture,
rather than a single, continuous cline of ancestry.
At all values of $K>1$, discrete population structure was mostly partitioned along species lines; 
at values of $K$ above 2, further discrete substructure was inferred within the \textit{P. trichocarpa} samples, 
while \bals{} samples displayed no substructure up to $K=7$.
This lack of substructure within the sampled \bals{} populations is in agreement with results from 
\citet{keller_etal_2010}, who found that \bals{} in the region is characterized by relatively low diversity, 
and probably represents a recent, postglacial expansion.

A consistent split between layers within \tri{} fell along the ``no-cottonwood belt," 
a region along the central coast of British Columbia in which black cottonwood is absent. 
The no-cottonwood belt is hypothesized to divide the species' distribution 
into northern and southern groups, which, in a provenance test, 
were experimentally shown to display different phenotypes 
\citep[pathogen resistance][]{xie2009,xie2012}.  
At higher values of $K$, drainages at the southern tip of \tri{} sampling 
begin to split out into their own layers.  
\plr{cut this if we pick lower $K$? Or add a disclaimer?}
This may potentially be the result of introgression from Southern \textit{Populus} species, 
such as \textit{P. angustifolia} or \textit{fremontii} \citep{Zhou2012,geraldes_etal_2014}

At all values of $K$, the spatial model was preferred over the nonspatial model.
However, most of this signal seems to derive from the \textit{P. trichocarpa} samples:
covariance within predominantly \tri{} layers decays more quickly with distance
than within the majority \bals{} layer, as seen in Fig \ref{populus_sp_clst_covs}.
The most strongly supported model, given the current dataset, was XXXX.


\subsubsection*{Black bears}


%%%%%%%%%
\section*{Discussion}

The pattern of isolation by distance is ubiquitous in nature,
unsuprisingly, since mating opportunities are distance-limited in most species.
However,
this common pattern can flummox commonly used clustering methods, 
which must use a possibly large number of discrete clusters
to describe such continuous population differentiation .
Inference of these spurious clusters can have implications for 
downstream analysis and interpretation of genetic data, 
including conservation management decisions and 
identification of the genetic basis of adaptation or disease traits.

In this paper, we have presented a statistical framework, \texttt{conStruct}, for simultaneously 
modeling continuous and discrete patterns of population structure.
By employing the sensible default assumption
that relatedness ought to decay with distance, even within a population, 
we avoid erroneously ascribing population differentiation to discrete population clusters.
We also present a cross-validation approach for comparing between models, 
although we stress the limits of this or any model selection procedure for genetic clustering.

The method performs well on simulated data:
we accurately infer the admixture proportions used to simulate the data 
and accurately pick the simulating model as the best model using our cross-validation procedure.
Empirical applications of \texttt{conStruct} to a sample of North American poplars 
and a sample of North American black bears yield reasonable results, 
largely in agreement with previous studies in both systems (Geraldes, Puckett).

The proposed method combines the utility of model-based clustering algorithms 
with a biologically realistic model of isolation by distance.
We anticipate that \texttt{conStruct} will be useful for identifying populations 
and determining samples' ancestry in and across them, 
especially when the populations exhibit spatial patterns of relatedness.

\paragraph{Comparison to nonspatial model-based clustering}
Above, we \plr{hopefully?} showed that 
(a) the nonspatial conStruct model recapitulates results of 
other, commonly-used nonspatial clustering methods,
and 
(b) conStruct can concisely capture spatial structure, which is nearly always expected within clusters.
Given this, when should methods without spatial capability be used?
One advantage these have over conStruct is speed \plr{details?}.
The other main difference is the model of noise,
but it is unclear what effects this may have,
or indeed, which is closer to the truth.

\paragraph{Choosing the ``best" number of clusters}
Although we recognize the utility of choosing a single, ``best'' value of $K$, 
and using only that analysis to communicate results, 
we want to emphasize that there are both statistical and biological reasons 
why there is no true best value of $K$.
From a statistical perspective, the choice of best $K$ is always relative to the data in hand.
This means both that sampling different individuals or loci 
from the same populations may yield a different ``best" $K$, 
but also 
-- unless the data were generated under the model itself --
that the support for larger values of $K$ is likely to increase with increasing amounts of data
(although perhaps only logarithmically).
In the limit of infinite data, the best value of $K$ 
may be the number of individuals included in the dataset.
From a biological perspective, 
it is important to stress that patterns of relatedness between individuals and populations 
are shaped by complex spatial and hierarchical processes.
All individuals within a species (and indeed, all individuals across all species), 
are related to one another in some way, 
and summarizing that relatedness with a single value of $K$ may be reductive or misleading.
We therefore encourage users to perform analyses across different values of $K$ and 
observe which layers split out at what levels 
(effectively taking successively shallower cross-sections of the population phylogeny),
and also to take the results of the proposed cross-validation procedure with a large grain of salt.

\paragraph{Implications for management and conservation}
If isolation by distance is common, 
then a likely result of applying \texttt{conStruct} to existing data is that 
populations previously identified using nonspatial clustering methods 
will be collapsed into each other.  
This ``lumping" might better reflect biological reality, 
but may also have implications for management decisions and conservation policy, 
both of which are often predicated on the identification of discrete ``management units" (MUs) 
identified using genetic data \citep{Moritz1994,Waples_1998,Moritz_etal_2002}.

It is therefore important to emphasize that individuals sampled from the same \texttt{conStruct} cluster 
may be quite genetically diverged from one another, 
and that a \texttt{conStruct} cluster may still contain multiple distinct MUs.  
Alternatively, the collapse of multiple, previously identified MUs into a single \texttt{conStruct} cluster 
may reflect a biological reality that these populations are currently 
(or were recently) exchanging migrants, 
and thus might emphasize the importance of maintaining habitat corridors (ref) 
between demes, or of implementing an integrated conservation plan across multiple demes within a cluster.

\paragraph{Caveats and considerations}
There are a few important caveats to consider in the interpretation of \texttt{conStruct} results. 
First, we have modeled allelic covariance within a cluster as a spatial process.
Although there is flexibility built into the model about the shape of that covariance, 
inference may be misleading if the sampling geography departs radically from the way 
the sampled organisms experience (or have experienced) their landscape.
For example, if we were to run a \texttt{conStruct} analysis using geographic distances between 
sampled individuals of greenish warblers (Irwin ref) or \textit{Ensatina} salamanders (Wake ref)
--- two canonical examples of rings species --- 
we might get misleading results.
This is because distance between locations on either side of the species' distributions
(across the Tibetan plateau and the Central Valley, respectively) 
is not representative of the path traversed in the coalescent of a pair of alleles sampled at those locations.

A second caveat is that, in some instances, 
membership in the same cluster may not mean that samples are particularly related.
If covariance within a cluster decays sharply with distance, 
and the cluster-specific relatedness parameter $\phi^{(k)}$ is low, 
individuals at a far spatial remove may be in the same cluster but have very low pairwise relatedness.
It is possible that this is happening in Fig \ref{populus_sp_pies}\subref{populus_sp_pies:K3}. 
At $K=3$, the southernmost populations of \textit{P. trichocarpa} are in the green cluster, 
whose other neighbors are to the north, with an intervening group of populations in the red cluster, 
and at $K=4$, those southernmost samples split out and become their own yellow cluster.
Again, we encourage users to run analyses across multiple values of $K$, 
and to examine the spatial covariance functions within layers when interpreting results.
\gb{missing data?}

\paragraph{Extensions and future directions}
There are several ways in which the model described in this paper might be extended or improved.  
For example, we currently assume that all layers within a model are equally unrelated 
(a star population phylogeny, although the branches can have different lengths thanks to the $\phi^{(k)}$ parameter), 
similar to the F-model in (REF the right STRUCTURE paper).
However, we could extend the existing model by implementing 
a relatedness structure between the layers by, for example, 
estimating a population phylogeny between them \citep[e.g.][]{treemix}.

A second exciting direction is to model relatedness within a cluster as a spatiotemporal process, 
in which covariance decays both with distance in space and in time.  
As the number of genotyped historical or ancient samples increases, 
it is becoming possible to ask whether there is genetic continuity at a point in space across time, 
or whether populations are being replaced (yamnaya paper, slatkin and racimo, nielsen 2016, schraiber 2017).
However, we expect allele frequencies to change through time in a population, even without replacement, 
simply due to drift.
Therefore, a natural way to test for population replacement is to estimate the rates 
at which relatedness within a cluster decays with time in the same way we do in the current model with space, 
in which case discrete population structure in space is comparable to population replacement in time.

\section*{Acknowledgements}

We thank Marjorie Weber, William Wetzel, Mariah Meek, Yaniv Brandvain, and Matthew Stephens 
for invaluable comments on the method and manuscript, 
as well as Quentin Cronk, who provided input on the \textit{Populus} analyses, 
and the attendees at the 2017 SSE Meeting in Portland, OR, 
whose votes determined the name of the method.
This work was supported in part by 
the National Science Foundation under award number NSF \#1262645 (DBI) to PR and GC, 
the National Institute of General Medical Sciences of the National Institutes of Health under award numbers NIH RO1GM83098 and RO1GM107374 to GC,
and the National Science Foundation under award numbers NSF \# 1148897 and \# 1402725 to GB.

\newpage
\section*{Appendix}
\renewcommand{\theequation}{A\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thetable}{A\arabic{table}}
\setcounter{table}{0}
\renewcommand{\thefigure}{A\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thesection}{A\arabic{section}}
\setcounter{section}{0}


\section{Model rationale}
\subsection{Drift, admixture, and space}
\paragraph{Drift} 
We first provide a simple model of allele frequencies within a cluster. 
Imagine a sample $i$ that draws all of its ancestry from cluster $k$.
The allele frequency in sample $i$ at locus $\ell$, $F_{i,\ell}$ can be
written as the sum
\begin{equation}
F_{i,\ell} = \epsilon_{\ell} + \Delta^{(k)}_{\ell} +
\Delta^{(k,i)}_{\ell} + \Delta^{(i)}_{\ell}
\label{drift_terms_no_admix}
\end{equation}
The first term is the ancestral allele frequency $\epsilon_\ell$ shared by all samples; 
the second is the deviation from that ancestral frequency 
due to drift in the ancestral population of the $k$th cluster,
which is shared by all samples within the cluster. 
The third term is the deviation of the $i$th sample away from the $k$th cluster mean 
due to the spatial process of drift and migration within the cluster.
The final term is the deviation specific to the $i$th sample,
which captures drift not shared by all samples at the population level
(i.e., subpopulation-specific drift due to, e.g., inbreeding). 
We will assume that these four deviations are all uncorrelated with each other.

If we have two samples $i$ and $j$ drawn from cluster $k$, 
their covariance across loci will be 
\begin{equation}
\text{Var}(\epsilon) +  \text{Var}\left( \Delta^{(k)} \right) +
\text{Cov}(\Delta^{(k,i)}_{\ell},\Delta^{(k,j)}_{\ell}) + \delta_{i=j} \text{Var}\left( \Delta^{(i)} \right)\ \\
\end{equation}
where the quantity $\delta_{i=j}$ is an indicator variable that equals 1 when $i$ is equal to $j$ and 0 otherwise, 
as in Eqn. \eqref{cross_cluster_covariance}.

\paragraph{Admixture} 
The model above describes the simple case 
in which samples draw 100\% of their ancestry from only a single cluster each. 
To accommodate admixture between clusters, 
we model allele frequencies within samples as linear combinations of the population-specific 
drift terms across the clusters from which they draw ancestry.
To do so, we introduce a term $w^{(k)}_{i}$ that describes the
admixture proportion of sample $i$ in cluster $k$ ($\sum_{K}w^{(k)}_{i} = 1$),
which can be interpreted as the proportion of the genome in the $i$th
sample that came from the $k$th cluster 
(or the probability that an allele at a locus is drawn from $k$).
The allele frequency in the $i$th sample at the $\ell$th locus can therefore be written as:
\begin{equation}
F_{i,\ell} = \epsilon_{\ell} + \sum\limits_{K} w^{(k)}_{i}\left( 
  \Delta^{(k)}_{\ell} + \Delta^{(k,i)}_{\ell}\right) + \Delta^{(i)}_{\ell}	
\label{drift_terms_admix}
\end{equation}
and the covariance between $i$ and $j$ across loci is
\begin{equation}
\Omega_{i,j} = \text{Var}(\epsilon) + \sum\limits_K w^{(k)}_iw^{(k)}_j
\left(
  \text{Var}\left( \Delta^{(k)} \right) +
\text{Cov}(\Delta^{(k,i)}_{\ell},\Delta^{(k,j)}_{\ell}) 	\right) +
\delta_{i=j} \text{Var}(\Delta_i)
\label{admixed_spatial_cov}
\end{equation}

\paragraph{Space}
Under our nonspatial model, 
we assume that $\text{Cov}(\Delta^{(k,i)}_{\ell},\Delta^{(k,j)}_{\ell})=0$,
so that the only additional covariance between $i$ and $j$ 
(above that induced by a shared ancestral frequency at each locus)
is due to the drift in the ancestral population of their cluster 
(the variance of which is $\phi_k$). 
Under our spatial model we assume that some of the covariance in allele frequencies
between $i$ and $j$ decays as a function of the geographic distance
between the pair, $D_{i,j}$,
so that 
\begin{equation}
\text{Cov}(\Delta^{(k,i)}_{\ell},\Delta^{(k,j)}_{\ell}) = \alpha^{(k)}_0 \times \left(\text{exp} \left(  -(\alpha^{(k)}_D D_{i,j})^{\alpha^{(k)}_2}\right) \right)
\end{equation}
We note that this form is chosen for its flexibility, 
and not because it matches any explicit population genetic model of isolation by distance. 

\section{Allelic Covariance}\label{allelic_cov}
To see how to arrive at Eqn. \eqref{allelic_covariance},
pick a random locus and
let $A$ and $B$ be randomly drawn alleles at that locus from populations $i$ and $j$ respectively.
If $i=j$, then these are chosen \emph{without} replacement.
Suppose these are each coded as `0' or `1` (where `0' denotes a reference allele),
but we randomly ``flip'' this coding, so that we let $X=A$ and $Y=B$ with probability 1/2,
but otherwise we let $X=1-A$ and $Y=1-B$.
We then let $\widehat{\Omega}_{i,j} = \cov[X,Y]$ ; 
sampling without replacement makes the statistic insensitive to sample size,
and random allele flipping makes it independent of the choice of reference allele.
By conditioning on the flip,
and using the fact that $\E[X] = \E[Y] = 1/2$,
Eqn. \eqref{allelic_covariance} comes from the calculation that
\begin{align}
\cov[X,Y] &= \E[XY] - 1/4 = (1/2)\E[AB + (1-A)(1-B)] - 1/4\\\notag
&= \E[(A-1/2)(B-1/2)]
\end{align}
The factor of $\delta_{i,j}/(N_i-1)$ comes from the fact that if $i=j$,
then $\E[AB]$ is the probability that both alleles drawn without replacement are reference,
which is $f_i^2 n/(n-1)$.
\paragraph{Likelihood}
If our allele frequency deviations are well approximated by a Gaussian, 
their sample allelic covariance is a sufficient statistic,
so that calculating the likelihood of their sample allelic covariance is the same as 
calculating the probability of the frequency data up to a constant. 
We can therefore model the covariance of the sample allele frequencies, $\widehat{\Omega}$, 
as a draw from a Wishart distribution with degrees of freedom equal to 
the number of loci $L$ across which the sample allelic covariance is calculated:
\gb{should we take this equation out?}
\begin{align}
&\widehat{\Omega}_{i,j} = 
	\begin{cases}
		\frac{1}{L-1}
		\sum\limits_{\ell=1}^{L}{\left[ 	\left(f_{i,\ell} - \bar{f}_{i} \right)
		 	\left(f_{j,\ell} - \bar{f}_{j} \right)\right] \notag
			} & \text{for } i \neq j \\
		\bar{f}_i (1-\bar{f}_i) & \text{for } i = j \\
	\end{cases}\\
&\widehat{\Omega} \sim \mathcal{W}\left( L\Omega, L	\right)
\label{wishart}
\end{align}

A benefit of directly modeling the sample allelic covariance is that, 
after the initial calculation of the sample covariance matrix,
the computation time of the likelihood is not a function of the number of loci,
so inference can theoretically be done using whole genome data.

\section{Models, parameters, and priors}\label{model_app}
\paragraph{Spatial vs. nonspatial}
In this paper, we discuss two types of models, spatial and nonspatial, 
each of which can be implemented with different numbers of clusters.
The spatial model is parameterized as in Eqn \eqref{admixed_spatial_cov},
and the nonspatial model is described in Eqn \eqref{admixed_discrete_covariance}.
The nonspatial model is a special case of the spatial model, 
in which the value of all $\alpha$ parameters is set to 0.
The nonspatial model therefore has $3K$ fewer parameters than the spatial model,
as there are three $\alpha$ parameters that describe the continuous differentiation effect of distance per cluster,
and they are fixed in the nonspatial model.

\paragraph{Single cluster}
Each of these models can be run with a single cluster ($K=1$), 
in which case the cluster-specific covariance parameter $\phi^{(k)}$ 
and the global covariance parameter $\gamma$ become redundant.
The single-cluster model is therefore a special case of the multi-cluster model, 
in which we set $\phi$ to zero, as described below:

For the spatial model, the single-cluster parametric covariance is:
\begin{equation}
\Omega_{i,j} = \text{Var}(\epsilon) + 
\alpha^{(k)}_0 \times \left(\text{exp} \left(  -(\alpha^{(k)}_D D_{i,j})^{\alpha^{(k)}_2}\right) \right)	 +
\delta_{i=j} \text{Var}(\Delta_i)
\label{admixed_spatial_cov}
\end{equation}

For the nonspatial model, the single-cluster parametric covariance is:
\begin{equation}
\Omega_{i,j} = \text{Var}(\epsilon) + \delta_{i=j} \text{Var}(\Delta^{(i)})
\label{admixed_discrete_covariance}
\end{equation}

\paragraph{Priors}
We use a Bayesian approach to parameter inference, 

A table of all parameters, their descriptions, and their priors is given in Table \ref{tab:param_prior_tab} below.

\begin{centering}
\begin{table}
\begin{tabular}{| >{\centering\arraybackslash}m{2.1cm} | m{5.2cm} | >{\centering\arraybackslash}m{5.1cm} |}
	\hline
	\textbf{Parameter} & \centering{\textbf{Description}} & \textbf{Prior}\\ \hline
	$\boldsymbol{\gamma}$ & 
		global covariance due to shared ancestral frequency & 
		$\gamma \sim \mathcal{N}(\mu = \text{Var}(\bar{f}), \sigma = 0.5)$\\ \hline
	$\boldsymbol{\alpha^{(k)}_0}$ & 
		controls the sill of the covariance matrix in cluster $k$& 
		$\alpha^{(k)}_0 \sim \mathcal{N}(\mu = 0, \sigma = 1)$\\ \hline
	$\boldsymbol{\alpha^{(k)}_D}$ & 
		controls the rate of the decay of covariance with distance in cluster $k$& 
		$\alpha^{(k)}_D \sim \mathcal{N}(\mu = 0, \sigma = 1)$\\ \hline
	$\boldsymbol{\alpha^{(k)}_2}$ & 
		controls the shape of the decay of covariance with distance in cluster $k$ & 
		$\alpha^{(k)}_2 \sim U(0,2)$\\ \hline
	$\boldsymbol{\eta_i}$ & 
		the nugget in population $i$ (population specific drift parameter)  & 
		$\eta_i \sim \mathcal{N}(\mu = 0, \sigma = 1)$\\ \hline
	$\boldsymbol{\phi^{(k)}}$ & 
		cluster-specific shared drift in cluster $k$ &
		 $\phi^{(k)} \sim \mathcal{N}(\mu = 0, \sigma = 1)$\\ \hline
	$\boldsymbol{w_i}$ &
		admixture proportions sample $i$ draws across $K$ clusters &
		$w_i \sim \text{Dir}(\alpha_{1} ... \alpha_{K}=0.1)$  \\ \hline
	\hline
\end{tabular}
\caption{
List of parameters used in the \texttt{conStruct} model, along with their descriptions and priors.
The mean of the Normal prior on $\gamma$, $\text{Var}(\bar{f})$, is the variance of the sample mean allele frequencies across loci.
}\label{tab:param_prior_tab}
\end{table}
\end{centering}

\newpage
\section{Cross validation}\label{Xvalidation}
To perform model comparison, we employ a Monte Carlo cross-validation approach \citep{picard1984},
also known as repeated random sub-sampling validation.
Briefly, we follow the following procedure:
\begin{enumerate}
\item For each of $X$ replicates:
	\begin{enumerate}
		\item partition the allele frequency data into a 90\% ``training'' partition ($F^x_1$) and a 10 \%``testing" partition ($F^x_2$) \label{partition}
		\item run our inference procedure on the training partition to estimate model parameters $\theta_{mk}$ for: \label{inference}
			\begin{enumerate}
				\item $m$: the spatial and the nonspatial model
				\item $k$: the number of clusters 1 through $K$
			\end{enumerate}	
	\item calculate the mean log likelihood of the testing data partition, 
	over the posterior distribution of training-estimated parameters for each model 
	($\bar{\mathcal{L}}(F^x_2 \mid \theta_{mk})$, henceforth $\bar{\mathcal{L}}_{xmk}$) \label{lnL}
	\item generate standardized mean log likelihoods, $\mathcal{Z}_{xmk}$, across models: \label{standardize}
		\begin{enumerate}
			\item identify the highest mean log likelihood, $\bar{\mathcal{L}}^\text{max}_{xmk}$
			\item subtract $\bar{\mathcal{L}}^\text{max}_{xmk}$ from $\bar{\mathcal{L}}_{xmk}$ across all models,
				such that the standardized log likelihood, $\mathcal{Z}_{xmk}$, of the best model is 0,
				and less than 0 for all inferior models. 
		\end{enumerate}
	\end{enumerate}
\item For each model (i.e., each combination of $m$ and $k$) calculate 
	the mean standardized log likelihood of the testing data partition across $X$ replicates, 
	as well its standard error and 95\% confidence interval:
	\begin{enumerate}
	\item mean
		\begin{equation}
			\bar{\mathcal{Z}}_{mk} = \frac{1}{X}\sum\limits_{x=1}^{X}\mathcal{Z}_{xmk}
		\end{equation}
	\item standard error
		\begin{equation}
			SE_{\bar{\mathcal{Z}}_{mk}} = \sqrt{\frac{\sum\limits_{x=1}^{X} \left[ {\mathcal{Z}_{xmk} - \bar{\mathcal{Z}}_{mk}} \right]}{X}}
		\end{equation}
	\item 95\% confidence interval
		\begin{equation}
			95\% \text{CI} = \bar{\mathcal{Z}}_{mk} \pm 1.96 \times SE_{\bar{\mathcal{Z}}_{mk}}
		\end{equation}
	\end{enumerate}
\end{enumerate}

This cross-validation procedure generates a mean predictive accuracy for each model and each value of $K$, 
as well as a confidence interval around that mean,
which can then be used for model comparison or selection.

\newpage
\section*{Supplementary Materials}
\renewcommand{\theequation}{S\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{table}{0}
\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}

\clearpage
\begin{figure}
	\centering
		{\includegraphics[width=4in,height=2.5in]{figs/sims/sim_setup.png}}
		\caption{Schematic of how we simulate datasets with continuous and discrete differentiation, using $K=2$ as an example.  
			    Going forward in time, the $K$ clusters split from a common ancestor at time $\tau_{\text{s}}$,
			    then each expand to colonize a lattice of demes with nearest-neighbor symmetric migration at time $\tau_{\text{e}}$,
			    then finally at time $\tau_{\text{a}}$ collapse into a single lattice consisting of demes 
			    with ancestry entirely in one or the other of the clusters,
			    or admixed between them.
			    }\label{sim_setup}
\end{figure}

\newpage
\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK1_nsp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK1_nsp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK1_nsp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK1_nsp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK1_nsp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK1_nsp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_nsp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a nonspatial model for $K=2$ through 7.
	The data were simulated using a single cluster with nearest-neighbor symmetric migration between demes.
    }\label{simK1_nsp_pies}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK1_sp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK1_sp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK1_sp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK1_sp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK1_sp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK1_sp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK1_sp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a spatial model for $K=2$ through 7.
	The data were simulated using 1 cluster with nearest-neighbor symmetric migration between demes.
    }\label{simK1_sp_pies}
\end{figure}

\newpage
\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK2_nsp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK2_nsp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK2_nsp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK2_nsp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK2_nsp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK2_nsp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_nsp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a nonspatial model for $K=2$ through 7.
	The data were simulated using two clusters with nearest-neighbor symmetric migration between demes.
    }\label{simK2_nsp_pies}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK2_sp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK2_sp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK2_sp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK2_sp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK2_sp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK2_sp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK2_sp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a spatial model for $K=2$ through 7.
	The data were simulated using two clusters with nearest-neighbor symmetric migration between demes.
    }\label{simK2_sp_pies}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK3_nsp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK3_nsp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK3_nsp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK3_nsp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK3_nsp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK3_nsp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_nsp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a nonspatial model for $K=2$ through 7.
	The data were simulated using three clusters with nearest-neighbor symmetric migration between demes.
    }\label{simK3_nsp_pies}
\end{figure}

\begin{figure}
	\centering
		\subcaptionbox{$K=2$\label{simK3_sp_pies:K2}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K2.pdf}}
		\subcaptionbox{$K=3$\label{simK3_sp_pies:K3}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K3.pdf}}
		\subcaptionbox{$K=4$\label{simK3_sp_pies:K4}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K4.pdf}}
		\subcaptionbox{$K=5$\label{simK3_sp_pies:K5}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K5.pdf}}
		\subcaptionbox{$K=6$\label{simK3_sp_pies:K6}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K6.pdf}}
		\subcaptionbox{$K=7$\label{simK3_sp_pies:K7}}
			{\includegraphics[width=1.8in,height=1.8in]{figs/sims/simK3_sp_pies_K7.pdf}}
	\caption{
	Map of admixture proportions estimated using a spatial model for $K=2$ through 7.
	The data were simulated using three clusters with nearest-neighbor symmetric migration between demes.
    }\label{simK3_sp_pies}
\end{figure}
\clearpage


\begin{figure}
	\centering
		{\includegraphics[width=6in,height=2.25in]{figs/fastStr/fastStr_simK1_pies.pdf}}
	\caption{
	Map of admixture proportions estimated using fastSTRUCTURE \citep{fastStructure} for $K=2$ through 4.
	The data were simulated using one cluster with nearest-neighbor symmetric migration between demes.
    }\label{fastStr_simK1}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=2.25in]{figs/fastStr/fastStr_simK2_pies.pdf}}
	\caption{
	Map of admixture proportions estimated using fastSTRUCTURE \citep{fastStructure} for $K=2$ through 4.
	The data were simulated using two clusters with nearest-neighbor symmetric migration between demes.
    }\label{fastStr_simK2}
\end{figure}

\begin{figure}
	\centering
		{\includegraphics[width=6in,height=2.25in]{figs/fastStr/fastStr_simK3_pies.pdf}}
	\caption{
	Map of admixture proportions estimated using fastSTRUCTURE \citep{fastStructure} for $K=2$ through 4.
	The data were simulated using three clusters with nearest-neighbor symmetric migration between demes.
    }\label{fastStr_simK3}
\end{figure}

\newpage
\clearpage
\bibliography{reference.library.tex.bib}

\newpage

\pagenumbering{gobble}

\end{document}
